lib/client/client.go

```diff
@@ -321,6 +326,37 @@ func (proxy *ProxyClient) GetAppServers(ctx context.Context, namespace string) (
 	return servers, nil
 }
 
+// GetDatabaseServers returns all registered database proxy servers.
+func (proxy *ProxyClient) GetDatabaseServers(ctx context.Context, namespace string) ([]services.DatabaseServer, error) {
+	authClient, err := proxy.CurrentClusterAccessPoint(ctx, false)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	servers, err := authClient.GetDatabaseServers(ctx, namespace, services.SkipValidation())
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return servers, nil
+}
+
+// GetDatabaseServersFor returns all servers proxying the specified database.
+func (proxy *ProxyClient) GetDatabaseServersFor(ctx context.Context, namespace, dbName string) (result []services.DatabaseServer, err error) {
```

Comment:
Since the filtering isn't happening server side, why not just the filtering at the caller?